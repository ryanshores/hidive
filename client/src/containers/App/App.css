.App {
	overflow: hidden;
}

.container-fluid {
	max-width: 1170px;
}

.content {
	color: #ccc;
	margin-top: 70px;
	background: #222325;
}

.section {
	margin: 0 15px;
}


.slick-slider-icon {
	padding: 10px;
	border-top: 4px solid rgb(90, 90, 90);
}

.slick-active .slick-slider-icon {
	border-color: rgba(0,174,240,1) !important;
}

@media (max-width: 600px) {
	.container-fluid {
		padding: 0;
	}
}