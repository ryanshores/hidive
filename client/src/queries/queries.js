import { gql } from 'apollo-boost';

const getRowQuery = gql`
	query($rowString: String!) {
		row(row: $rowString) {
			Id
			Name
			MasterArtUrl
		}
	}
`;

export { getRowQuery };
