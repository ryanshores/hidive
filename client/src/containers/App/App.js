import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { graphql } from 'react-apollo';
import './App.css';

import { getVideos } from '../../queries/queries';

import Header from '../../components/Header/Header';
import HomeList from '../VideoList/HomeList';
import VideoList from '../VideoList/VideoList';

class App extends Component {
  state = {
    titles: [],
    loading: true
  }

  // Checks if the props have changed and if the title object exists
  componentDidUpdate(prevProps) {
    if(this.props.data !== prevProps.data){
      if(this.props.data.titles) {
        this.setState({
          titles: this.props.data.titles,
          loading: false
        })
      }
    }
  }

  render() { 
    const { titles, loading } = this.state;
    const titlesObject = setupTitlesObject(titles);
    return (
      <Router className="Router">
        <div className="App">
          <Header />
          <Content 
            titlesObject={titlesObject}
            loading={loading}/>
        </div>
      </Router>
    );
  }
}

const Content = ({titlesObject, loading}) => {
  return (
    <div className="container-fluid fadeIn">
      <div className="content">
        { loading ? Loading() : Routes(titlesObject) }
      </div>
    </div>
  )
};

const Routes = (titlesObject) => {
  const keys = titlesKeys(titlesObject);
  const RoutesObjects = keys.map(key => (
    <Route 
      key={key}
      path={'/' + key} 
      render={() => <VideoList rowid={key} titles={titlesObject[key]}/>} />
  ));
  return (
    <div>
      <Route exact path='/' render={() => <HomeList titles={titlesObject}/>}/>
      { RoutesObjects }
    </div>
  )
};

const Loading = () => {
  return (
    <div className='load'>
      <div className="line"></div>
      <div className="line"></div>
      <div className="line"></div>
    </div>
  )
};

const titlesKeys = (titlesObejct) => Object.keys(titlesObejct);

const setupTitlesObject = (titles) => ({
  simulcasts: titles.filter(title => title.isSimulcast),
  dubs: titles.filter(title => title.isDubed),
  exclusive: titles.filter(title => title.isExclusive),
  recent: titles.filter(title => title.isRecent),
  trending: titles.filter(title => title.isTrending),
  popular: titles.filter(title => title.isPopular),
  movies: titles.filter(title => {
    if(
      title.ShowInfoTitle.indexOf('Theatrical') === 0 ||
      title.ShowInfoTitle.indexOf('OVA') === 0 ||
      title.ShowInfoTitle.indexOf('Special') === 0
    ) {
      return true
    } else {
      return false
    }
  }),
  series: titles.filter(title => {
    if(
      title.ShowInfoTitle.indexOf('Season') === 0 ||
      title.ShowInfoTitle.indexOf('Shorts') === 0 ||
      title.ShowInfoTitle.indexOf('Knight') === 0 ||
      title.ShowInfoTitle.indexOf('Rondo') === 0
    ) {
      return true
    } else {
      return false
    }
  }),
  continueWatching: titles.filter(title => title.IsContinueWatching),
})
 
export default graphql(getVideos)(App);