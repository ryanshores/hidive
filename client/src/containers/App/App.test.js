import React from 'react';
import { create as render } from 'react-test-renderer';
import '../../setupTest';
import { MockedProvider } from 'react-apollo/test-utils';

import { getTitles } from '../../queries/queries'
import App from './App';

const mocks = [
	{
		request: {
			query: getTitles
		},
		result: {
			data: {
				titles: [
					{
						Id: "802",
						IsContinueWatching: true,
						MasterArtUrl: "//d10xkldqejj5hr.cloudfront.net/titles/YYP/YYP_01_MASTER_300x169.jpg",
						Name: "Yona Yona Penguin",
						ShowInfoTitle: "Theatrical | G",
						isDubed: true,
						isExclusive: true,
						isPopular: false,
						isRecent: true,
						isSimulcast: false,
						isTrending: false,
						__typename: "Title",
					}
				]
			}
		}
	},
]

it('renders without crashing', () => {
	const component = render(
		<MockedProvider mocks={mocks} addTypename={false}>
			<App />
		</MockedProvider>
	);
	const snap = component.toJSON();
	expect(snap).toMatchSnapshot();
})