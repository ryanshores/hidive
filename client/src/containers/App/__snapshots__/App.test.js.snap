// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg fixed-top navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <button
        className="navbar-toggler"
        data-target="#links"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="HiDive"
          src="http://d10xkldqejj5hr.cloudfront.net/content/images/HIDIVE_logo.svg"
          width="140"
        />
      </a>
      <div
        className="collapse navbar-collapse"
        id="links"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item active"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/simulcasts"
              onClick={[Function]}
            >
              simulcasts
            </a>
          </li>
          <li
            className="nav-item active"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/dubs"
              onClick={[Function]}
            >
              dubs
            </a>
          </li>
          <li
            className="nav-item active"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/movies"
              onClick={[Function]}
            >
              movies
            </a>
          </li>
          <li
            className="nav-item active"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/series"
              onClick={[Function]}
            >
              series
            </a>
          </li>
        </ul>
        <div
          className="searchContainer ml-auto"
        >
          <form
            className="form-inline my-2 my-lg-0"
            onSubmit={[Function]}
          >
            <input
              aria-label="Search"
              className="form-control mr-sm-2"
              name="search"
              onChange={[Function]}
              placeholder="Search"
              type="search"
              value=""
            />
          </form>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container-fluid fadeIn"
  >
    <div
      className="content"
    >
      <div
        className="load"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;
